docs/backend-requests/making/jwt-templates.mdx

```diff
@@ -160,7 +160,65 @@ For example, you could use interpolation to build the user's full name:
 }
 ```
 
-Interpolated shortcode substituted values will always be casted to strings. For example, if the user does not have a last name associated, the above full name value would be "null John".
+Interpolated shortcodes will always result to string values. For example, if the user does not have a last name associated, the above full name value would be "null John".
+
+### Coalescing
+
+Shortcode coalescing is a conditional expression of the form:
+
+```json
+"{{ <operand_1> || <operand_2> || <operand_n> }}"
+```
+
+For example, given the following template:
+
+```json
+{
+  "has_verified_contact_info": "{{user.has_verified_email || user.has_verified_phone}}",
+
+  // fallback to a string value
+  "full_name": "{{user.first_name || 'John'}} {{user.last_name || 'Doe' }}",
+
+  // fallback to a number value
+  "age": "{{user.public_metadata.age || user.unsafe_metadata.age || 30 }}"
+}
+```
+
+The output after substitution (subject to the actual user data) would look like this:
+
+```json
+{
+  "has_verified_contact_info": true,
+  "full_name": "Mary Doe",
+  "age": 30
+}
+```
+
+Coalescing is often used to substitute a default fallback value for shortcodes that would otherwise result in `null` or `false` values.
+
+In a coalescing expression, operands are separated by `||`. The result of a coalescing expression is the first operand that is not `null`/`false`. If all operands of the expression are `null`/`false`, the last operand is returned no matter its return value.
+
+Coalescing can be combined with interpolation:
+
+```json
+{
+  "title": "Awesome {{user.first_name || user.last_name || 'user'}}"
+}
+```
+
+Assuming a user that has no first or last name set in their profile, the above would output:
+
+```json
+{
+  "title": "Awesome user"
+}
+```
+
+The rules that sum up how coalescing works are:
+
+- string operands should use single quotes (`'`)
```

Comment:
❓ Is this a requirement though? It's definitely good to advise that though.

Reply:
it is a requirement due to https://github.com/clerkinc/clerk_go/pull/4441/files#r1356754898

Reply:
Are you sure? My understanding is that quotes are used to enclose literal values that may contain characters that would otherwise confuse the parser. So for example:
- `user.public_metadata.missing_key || hello || world` will return `hello` instead of `hello || world`
- leading/trailing whitespace will be omitted.

Reply:
In your example (i.e. not using any quotes to surround the string), we would get `"hello"` indeed.

However, if for example you do:

```json
"{{ user.public_metadata.missing_key || \"hello\" }}"
```

you're going to get  `"{{ user.public_metadata.missing_key || \"hello\" }}"`, not  `"hello"`. Just tried this using one of the test cases.